src/app/catalog/components/badge-catalog/badge-catalog.component.ts

```diff
@@ -172,25 +201,24 @@ export class BadgeCatalogComponent extends BaseRoutableComponent implements OnIn
 		};
 	}
 
-	changeOrder(order) {
-		this.order = order;
-		if (this.order === 'asc') {
-			this.badgeResults.sort((a, b) => a.name.localeCompare(b.name));
-			this.badgeResultsByIssuer
-				.sort((a, b) => a.issuerName.localeCompare(b.issuerName))
-				.forEach((r) => r.badges.sort((a, b) => a.name.localeCompare(b.name)));
-			this.badgeResultsByCategory
-				.sort((a, b) => a.category.localeCompare(b.category))
-				.forEach((r) => r.badges.sort((a, b) => a.name.localeCompare(b.name)));
-		} else {
-			this.badgeResults.sort((a, b) => b.name.localeCompare(a.name));
-			this.badgeResultsByIssuer
-				.sort((a, b) => b.issuerName.localeCompare(a.issuerName))
-				.forEach((r) => r.badges.sort((a, b) => b.name.localeCompare(a.name)));
-			this.badgeResultsByCategory
-				.sort((a, b) => b.category.localeCompare(a.category))
-				.forEach((r) => r.badges.sort((a, b) => b.name.localeCompare(a.name)));
-		}
+	changeOrder(sortOption: string) {
+		const [sortBy, order] = sortOption.split('_') as ['name' | 'date', 'asc' | 'desc'];
+		const multiplier = order === 'asc' ? 1 : -1;
+
+		// Sortierfunktion
+		const sortFn = (a: any, b: any) => {
+			if (sortBy === 'name') {
+				return multiplier * a.name.localeCompare(b.name);
+			} else if (sortBy === 'date') {
+				const dateA = new Date(a.createdAt).getTime();
+				const dateB = new Date(b.createdAt).getTime();
+				return multiplier * (dateA - dateB);
+			}
+			return 0;
```

Comment:
This case should be impossible, right? Maybe throw an exception or at least log something here then?